src/lib/run-studies.ts

```diff
@@ -1,111 +1,64 @@
 /* eslint-disable no-console */
-import {
-    ECSClient,
-    LaunchType,
-    RunTaskCommand,
-    RunTaskCommandInput,
-    RegisterTaskDefinitionCommand,
-    RegisterTaskDefinitionCommandInput,
-    DescribeTaskDefinitionCommand,
-    DescribeTaskDefinitionCommandOutput,
-    RunTaskCommandOutput,
-} from '@aws-sdk/client-ecs'
-import { GetResourcesCommand, ResourceGroupsTaggingAPIClient } from '@aws-sdk/client-resource-groups-tagging-api'
+import { ECSClient, RunTaskCommandOutput } from '@aws-sdk/client-ecs'
+import { ResourceGroupsTaggingAPIClient } from '@aws-sdk/client-resource-groups-tagging-api'
+import { getECSTaskDefinition, registerECSTaskDefinition, runECSFargateTask, getTaskResourcesByRunId } from './aws'
 import { managementAppGetRunnableStudiesRequest, toaGetRunsRequest, filterManagmentAppRuns } from './utils'
 import 'dotenv/config'
 
 async function launchStudy(
     client: ECSClient,
     cluster: string,
     baseTaskDefinition: string,
-    subnets: string,
+    subnets: string[],
     securityGroup: string,
     runId: string,
     imageLocation: string,
     studyTitle: string,
 ): Promise<RunTaskCommandOutput> {
-    const baseTaskDefinitionData = await getTaskDefinition(client, cluster, baseTaskDefinition)
-    const containerDefinition = baseTaskDefinitionData.taskDefinition?.containerDefinitions?.map((container) => {
-        return {
-            ...container,
-            image: imageLocation,
-        }
-    })
-
     const taskTags = [
         { key: 'runId', value: runId },
         { key: 'title', value: studyTitle },
```

Comment:
Are study titles used?

Reply:
Good question / point: They are not used by the code like `runId`, but KC and I thought it might be nice to tag things with a "human readable" value in case it helps devs when debugging / investigating issues. So, it's there purely as a quality-of-life thing. If we find / think it's not useful, we can of course remove it.

Reply:
Makes sense. we can keep it, it's fine by me.